### DISCUSSION TITLE
Determining source IP address and port in connect()

### DISCUSSION THREAD

**Discussion by ytrewq271828** [ANLAB-KAIST/KENSv3#178] [Project Q&A 2. Transport Layer (TCP)]
Hello. 
During implementing connect(), I've utilized the routing functions like the following.
![image](https://github.com/ANLAB-KAIST/KENSv3/assets/52207623/75f96b7c-d821-49d1-8db4-7d8f0e76200a)

However, on the server side, the arrived port number of client is always 0, regardless of endianness.
I want to know whether it is a correct way to determine source IP address and port number.

Thanks in advance.

**Comment by Pusnow** [comment]
In KENS, there are two kinds of ports.

1. Port number in TCP header.
2. *Physical* port in NIC.

The `getRoutingTable` function is used to retrieve the physical port in NIC (2), so the return value is always zero if there is only one port in NIC.

To retrieve the port numbers in TCP header, you should parse the packet's TCP header and also store the parameter from `bind()`.
